jmx/plugins/org.jboss.tools.jmx.jolokia/src/org/jboss/tools/jmx/jolokia/internal/connection/JolokiaMBeanServerConnection.java

```diff
@@ -246,39 +241,86 @@ public Object invoke(ObjectName name, String operationName, Object[] params, Str
 		}
 	}
 
+	private J4pExecRequest createJ4pExecRequest(ObjectName name, Object[] params, String operationNameWithSignature) {
+		if(params == null || params.length == 0){
+			return new J4pExecRequest(name, operationNameWithSignature);
+		} else {
+			return new J4pExecRequest(name, operationNameWithSignature, params);
+		}
+	}
 
-	
-	
-	/*
-	 * The following methods are *CURRENTLY* unsupported due to missing classname in json response. 
-	 */
-	
+	private String createOperationNameWithSignature(String operationName, String[] signature) {
+		StringJoiner stringJoiner = new StringJoiner(",", "(", ")");
+		Stream.of(signature).forEach(stringJoiner::add);
+		return operationName + stringJoiner.toString();
+	}
 
 	@Override
 	public ObjectInstance getObjectInstance(ObjectName name) throws InstanceNotFoundException, IOException {
-		// TODO Auto-generated method stub
-		return null;
+		return createObjectInstance(name);
 	}
 
 	@Override
-	public boolean isInstanceOf(ObjectName name, String className) throws InstanceNotFoundException, IOException {
-		// TODO Auto-generated method stub
-		return false;
-	}
-	@Override
 	public Set<ObjectInstance> queryMBeans(ObjectName name, QueryExp query) throws IOException {
-		// TODO Auto-generated method stub
-		return Collections.emptySet();
+		Set<ObjectInstance> res = new HashSet<>();
+		try {
+			J4pSearchRequest req = new J4pSearchRequest(name.getCanonicalName());
+			J4pResponse<J4pSearchRequest> j4pResponse = j4pClient.execute(req);
+			Object value = j4pResponse.getValue();
+			if(value instanceof JSONArray){
+				for (Object mbean : (JSONArray)value) {
+					if(mbean instanceof String){
+						res.add(createObjectInstance((String)mbean));
+					}
+				}
+			}
+		} catch (MalformedObjectNameException | J4pException e) {
+			Activator.pluginLog().logError(e);
+		}
+		return res;
+	}
+
+	private ObjectInstance createObjectInstance(String mbean) throws MalformedObjectNameException {
+		ObjectName objectName = new ObjectName(mbean);
+		return createObjectInstance(objectName);
+	}
+
+	private ObjectInstance createObjectInstance(ObjectName objectName) {
+		String classname = retrieveClassName(objectName);
+		return new ObjectInstance(objectName, classname);
+	}
+
+	private String retrieveClassName(ObjectName objectName) {
+		J4pListRequest listAttributes = new J4pListRequest(objectName.getDomain()+"/"+objectName.getCanonicalKeyPropertyListString()+"/class");
+		try {
+			J4pResponse<J4pListRequest> listAttributesResponse = j4pClient.execute(listAttributes);
+			return listAttributesResponse.getValue();
+		} catch (J4pException e) {
+			// TODO : can we avoid to catch an exception? How to know if the class attribute need to be called or not?
```

Comment:
The exception is because of "/" included in the MBeanName, which needs to be escaped when calling it via HTTP GET as the slash is used to separate the various parts.

For robustness sake I highly recommend to use POST for Jolokia requests. You can switch this on per request and afair also per J4pClient.